### PR TITLE
invoices: fix SQL invoice query pagination

### PR DESCRIPTION
**Pull request by bhandras**
Previously paginated queries offseted the add_index_get, add_index_let, settle_index_get and settle_index_let parameters with the paginators current page offset, however this was incorrect as we can just use SQL's LIMIT/OFFSET to paginate. This commit fixes this issue and adds an optional parameter to the constructor of the invoice SQL store to set page size. This is useful when testing as we can now cover pagination correctly with our existing unit tests.

### CODE REVIEW

**Review by ProofOfKeags — APPROVED** [review]
🫡

File: invoices/invoiceregistry_test.go [hunk 1]
```diff
@@ -24,12 +23,6 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// sqliteConstructorMu is used to ensure that only one thread can call the
-// sqldb.NewTestSqliteDB constructor at a time. This is a temporary workaround
```
Comment (ellemouton) [inline invoices/invoiceregistry_test.go:28]: yay

File: go.mod [hunk 2]
```diff
@@ -203,6 +203,9 @@ replace github.com/gogo/protobuf => github.com/gogo/protobuf v1.3.2
 // allows us to specify that as an option.
 replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-display v1.30.0-hex-display
 
+// Temporary replace until the next version of sqldb is tagged.
+replace github.com/lightningnetwork/lnd/sqldb => ./sqldb
```
Comment (ellemouton) [inline go.mod:207]: cool - not sure we want to merge with this thought right? 

So I think might need to open separate PR that points to this (so we can verify complete CI run) and then once this PR is merged and tagged, then update that PR to point to the tagged version

**Review by ellemouton — APPROVED** [review]
LGTM! nice catch ⚾ 

Think we may want to open a second PR to do the go.mod change though so that we dont need to merge the temp replace

File: go.mod (hunk 2 above)
Comment (bhandras) [reply go.mod:207]: So the next two migration PRs both touch sqldb and other packages depending on it so i think we can't avoid. Maybe can move this to the next PR (#8855) though.

**Pull request merged**